src/evg_config.rs

```diff
@@ -0,0 +1,848 @@
+    fn lookup_required_param_u64(&self, task_def: &EvgTask, run_var: &str) -> Result<u64> {
+        Ok(match self.get_gen_task_var(task_def, run_var) {
+            Some(v) => v.parse()?,
+            _ => bail!(format!(
+                "Missing var '{}' for task '{}'",
+                run_var, task_def.name
+            )),
+        })
+    }
+
+    /// Lookup the given variable in the task definition.
+    ///
+    /// # Arguments
+    ///
+    /// * `task_def` - Task definition to query.
+    /// * `run_var` - Variable to query.
+    ///
+    /// # Returns
+    ///
+    /// Value of run_var for the given task, an `Error` will be returned if not defined.
+    fn lookup_required_param_bool(&self, task_def: &EvgTask, run_var: &str) -> Result<bool> {
+        Ok(match self.get_gen_task_var(task_def, run_var) {
+            Some(v) => v.parse()?,
+            _ => bail!(format!(
+                "Missing var '{}' for task '{}'",
+                run_var, task_def.name
+            )),
+        })
+    }
+
+    /// Lookup the given variable in the task definition.
+    ///
+    /// # Arguments
+    ///
+    /// * `task_def` - Task definition to query.
+    /// * `run_var` - Variable to query.
+    /// * `default` - Default value to use if `run_var` is undefined.
+    ///
+    /// # Returns
+    ///
+    /// Value of run_var for the given task, the default will be returned if not defined.
+    fn lookup_default_param_bool(
+        &self,
+        task_def: &EvgTask,
+        run_var: &str,
+        default: bool,
+    ) -> Result<bool> {
+        Ok(match self.get_gen_task_var(task_def, run_var) {
+            Some(v) => v.parse()?,
+            _ => default,
+        })
+    }
+
+    /// Lookup the given variable in the task definition.
+    ///
+    /// # Arguments
+    ///
+    /// * `task_def` - Task definition to query.
+    /// * `run_var` - Variable to query.
+    /// * `default` - Default value to use if `run_var` is undefined.
+    ///
+    /// # Returns
+    ///
+    /// Value of run_var for the given task, the default will be returned if not defined.
+    fn lookup_default_param_str(&self, task_def: &EvgTask, run_var: &str, default: &str) -> String {
+        self.get_gen_task_var(task_def, run_var)
+            .unwrap_or(default)
+            .to_string()
+    }
+}
+
+/// Get the shrub function make the 'generate resmoke task' call in the given task.
+///
+/// # Arguments
+///
+/// * `task` - Shrub task to query.
+///
+/// # Returns
+///
+/// Function call to 'generate resmoke task'.
+fn get_generate_resmoke_func(task: &EvgTask) -> Option<&FunctionCall> {
+    let command = task.commands.iter().find(|c| {
+        if let Function(func) = c {
+            if func.func == GENERATE_RESMOKE_TASKS {
+                return true;
+            }
+        }
+        false
+    });
+
+    if let Some(Function(func)) = command {
+        Some(func)
+    } else {
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
This create a new module. Since it is tagged with `#[cfg(test)]`, rust knows that it is just test code and will strip it out of the binary.

Reply:
Does rust usually advocate for putting the tests for a module in the same file as the code it is testing? It seems to lead to pretty large file sizes

Reply:
Yes, I'm following the conventions from [here](https://doc.rust-lang.org/book/ch11-03-test-organization.html). This file in particular is doing a lot, I think we can split it up.